### DISCUSSION TITLE
Floating IP & Cert-manager

### DISCUSSION THREAD

**Discussion by mashkovd** [kube-hetzner/terraform-hcloud-kube-hetzner#829] [Q&A] [answered]
Hello

If I use this ip 
![image](https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/assets/1842423/de495d20-e6ae-4a08-8b37-eac842c70c99)
when configure DNS and get Certificate by Cert-manager - all is OK
But when i create floating IP, configure DNS with this Floating IP and conect Floating IP and server
![image](https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/assets/1842423/831572ff-2af3-4af9-af29-81f3226ec65e)
i have problem with challenge process

Status:
  Presented:   true
  Processing:  true
  Reason:      Waiting for HTTP-01 challenge propagation: failed to perform self check GET request 'http://site.com/.well-known/acme-challenge/b0KNZzzwCc2-o3HrTkIgilfdYBh71yTBN1PDu5rSNmk': Get "http://site.com/.well-known/acme-challenge/b0KNZzzwCc2-o3HrTkIgilfdYBh71yTBN1PDu5rSNmk": dial tcp 116.202.7.239:80: connect: no route to host
  
  Do you meet with problem?
thx

**Comment by mashkovd** [comment]
it was some docs https://docs.hetzner.com/cloud/floating-ips/persistent-configuration/

but i use OpenSUSE MicroOS ARM by Kube-Hetzner

**Reply by mysticaltech** [reply]
@mashkovd Try with DNS challenge.

**Reply by mashkovd** [reply, marked as answer]
Problem is not DNS. I associate floating ip and cname.
It's work after i do this "sudo ip addr add ***.202.7.***  dev eth0" this command on server
***.202.7.*** - floating IP.
Instruction [floating-ips](https://docs.hetzner.com/cloud/floating-ips/persistent-configuration/) is only for Ubuntu 20.04 (not for OpenSUSE MicroOS)

**Reply by mysticaltech** [reply]
Interesting, PR welcome and appreciated to fix this missing behavior when using floating IPs.

**Reply by mysticaltech** [reply]
I had no idea, so we probably need to edit at the level of cloud-init template.
![ksnip_20230605-112631](https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/assets/518555/29493ea0-6790-4926-9cb3-98dbe0214cc1)

**Comment by sshcherbinin** [comment]
[maybe this will help](https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/pull/611)
already in the main stream
